packages/wallet/src/mock-rollup-state-machine.ts

```diff
@@ -0,0 +1,209 @@
+/* External Imports */
+
+/* Internal Imports */
+import {
+  Address,
+  Balances,
+  Swap,
+  isSwapTransaction,
+  Transfer,
+  isTransferTransaction,
+  Transaction,
+  MockedSignature,
+  SignedTransaction,
+  TransactionReceipt,
+  UNISWAP_ADDRESS,
+  UNI_TOKEN_TYPE,
+  PIGI_TOKEN_TYPE,
+  TokenType,
+  State,
+} from '.'
+
+const DEFAULT_STORAGE = {
+  balances: {
+    uni: 0,
+    pigi: 0,
+  },
+}
+
+/*
+ * Errors
+ */
+export class SlippageError extends Error {
+  constructor() {
+    super('Too much slippage in swap tx!')
+  }
+}
+
+export class InsufficientBalanceError extends Error {
+  constructor() {
+    super('Insufficient balance for transfer or swap!')
+  }
+}
+
+export class NegativeAmountError extends Error {
+  constructor() {
+    super('Amounts transferred or swapped cannot be negative!')
+  }
+}
+
+export class InvalidTransactionTypeError extends Error {
+  constructor() {
+    super('Invalid transaction type!')
+  }
+}
+
+export class MockRollupStateMachine {
+  public state: State
+
+  constructor(genesisState: State) {
+    this.state = genesisState
+  }
+
+  public getBalances(account: Address): Balances {
+    if (!(account in this.state)) {
+      return {
+        uni: 0,
+        pigi: 0,
+      }
+    }
+    return this.state[account].balances
+  }
+
+  public getUniswapBalances(): Balances {
+    return this.state[UNISWAP_ADDRESS].balances
+  }
+
+  private ecdsaRecover(signature: MockedSignature): Address {
+    // TODO: Move this out of this class and instead put in keystore
+    return signature
+  }
+
+  public applyTransaction(
+    signedTransaction: SignedTransaction
+  ): TransactionReceipt {
+    const sender: Address = signedTransaction.signature
+    const transaction: Transaction = signedTransaction.transaction
+    if (isTransferTransaction(transaction)) {
+      return this.applyTransfer(sender, transaction)
+    } else if (isSwapTransaction(transaction)) {
+      return this.applySwap(sender, transaction)
+    }
+    throw new InvalidTransactionTypeError()
+  }
+
+  private getTxReceipt(stateUpdate: any): TransactionReceipt {
+    return {
+      aggregatorSignature: 'MOCKED',
+      stateUpdate,
+    }
+  }
+
+  private hasEnoughMoney(
```

Comment:
Would change to `hasBalance(...)`, and change `greaterThanAmount` to `balance` since it's `>=` not `>`